pkg/controller/container-runtime-config/container_runtime_config_controller_test.go

```diff
@@ -331,31 +332,49 @@ func (f *fixture) expectUpdateContainerRuntimeConfig(config *mcfgv1.ContainerRun
 	f.actions = append(f.actions, core.NewRootUpdateSubresourceAction(schema.GroupVersionResource{Version: "v1", Group: "machineconfiguration.openshift.io", Resource: "containerruntimeconfigs"}, "status", config))
 }
 
-func (f *fixture) verifyRegistriesConfigContents(t *testing.T, mcName string, imgcfg *apicfgv1.Image, icsp *apioperatorsv1alpha1.ImageContentSourcePolicy) {
+func (f *fixture) verifyRegistriesConfigAndPolicyJSONContents(t *testing.T, mcName string, imgcfg *apicfgv1.Image, icsp *apioperatorsv1alpha1.ImageContentSourcePolicy, numFiles int) {
 	icsps := []*apioperatorsv1alpha1.ImageContentSourcePolicy{}
 	if icsp != nil {
 		icsps = append(icsps, icsp)
 	}
 	updatedMC, err := f.client.MachineconfigurationV1().MachineConfigs().Get(mcName, metav1.GetOptions{})
 	require.NoError(t, err)
-	verifyRegistriesConfigContents(t, updatedMC, mcName, imgcfg, icsps)
+	verifyRegistriesConfigAndPolicyJSONContents(t, updatedMC, mcName, imgcfg, icsps, numFiles)
 }
 
-func verifyRegistriesConfigContents(t *testing.T, mc *mcfgv1.MachineConfig, mcName string, imgcfg *apicfgv1.Image, icsps []*apioperatorsv1alpha1.ImageContentSourcePolicy) {
+func verifyRegistriesConfigAndPolicyJSONContents(t *testing.T, mc *mcfgv1.MachineConfig, mcName string, imgcfg *apicfgv1.Image, icsps []*apioperatorsv1alpha1.ImageContentSourcePolicy, numFiles int) {
 	// This is not testing updateRegistriesConfig, which has its own tests; this verifies the created object contains the expected
 	// configuration file.
 	expectedRegistriesConf, err := updateRegistriesConfig(templateRegistriesConfig,
 		imgcfg.Spec.RegistrySources.InsecureRegistries,
 		imgcfg.Spec.RegistrySources.BlockedRegistries, icsps)
 	require.NoError(t, err)
-
 	assert.Equal(t, mcName, mc.ObjectMeta.Name)
-	require.Len(t, mc.Spec.Config.Storage.Files, 1)
-	file := mc.Spec.Config.Storage.Files[0]
-	assert.Equal(t, registriesConfigPath, file.Node.Path)
-	registriesConf, err := dataurl.DecodeString(file.Contents.Source)
+	require.Len(t, mc.Spec.Config.Storage.Files, numFiles)
+	regfile := mc.Spec.Config.Storage.Files[0]
+	if regfile.Node.Path != registriesConfigPath {
+		regfile = mc.Spec.Config.Storage.Files[1]
+	}
+	assert.Equal(t, registriesConfigPath, regfile.Node.Path)
+	registriesConf, err := dataurl.DecodeString(regfile.Contents.Source)
 	require.NoError(t, err)
 	assert.Equal(t, string(expectedRegistriesConf), string(registriesConf.Data))
+
+	// Validate the policy.json contents if a change is expected from the tests
+	if numFiles == 2 {
+		expectedPolicyJSON, err := updatePolicyJSON(templatePolicyJSON,
```

Comment:
Please add actual tests for `updatePolicyJSON`, then — at least a smoke test for “no blocked/no allowed”, “something blocked”, “something allowed”.

Reply:
Added tests to helpers_test.go